medusa/common.py

```diff
@@ -559,7 +559,7 @@ def should_replace(ep_status, old_quality, new_quality, allowed_qualities, prefe
         :param search_type: The search type, that started this method
         :return: True if the old quality should be replaced with new quality
         """
-        if ep_status and ep_status not in (DOWNLOADED, SNATCHED, SNATCHED_PROPER):
```

Comment:
I suspect some users have `quality` as `0` for some reason and this was causing it to skip the second condition.